### PR TITLE
SJSU-1021-2: Connected Backend Code to Landing Redux

### PR DESCRIPTION
**Pull request by DaleCS**
What I did:
Connected backend code for Landing component to Landing action-redux workflow. 
I also removed some console.log(s) in the landing backend code.

### CODE REVIEW

File: front_end/src/components/landing_page/Landing.js [hunk 1]
```diff
@@ -129,34 +119,43 @@ class Landing extends Component {
...
+            return city;
+          }
+        });
+        return uniqueFeatureDestination;
+      }
+
+      // Uses genFeatureDestination() to create markup for each featured city.
+      let featureDestinationMarkup = genFeatureDestination().map(city => {
         return (
           <Grid item className="mouseHover" xs={3}>
```
Comment (SandroSal) [inline front_end/src/components/landing_page/Landing.js]: can we add the city name over the pic?

File: routes/api/landing.js [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    if (err) { 
+      res.status(400).json(err); 
+    }
+    var destinations = [];
+    var i;
+    //change loop limit if want more featured cities
+    var index = Math.floor(Math.random() * doc.length)
+    var headImg = doc[index].imgMain
+    var headCityName = doc[index].name
+    for (i = 0; i < 5; i++) {
```
Comment (SandroSal) [inline routes/api/landing.js]: are we going to have more than 4 cities?

File: routes/api/landing.js [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+      for(i = 0; i < 5; i++){
+          destinations.push(doc.pop());
+      }
+      res.send(destinations);
+      */
+    })
+    
+
+  //fisher yates shuffle of cities
+function shuffle(arr) {
```
Comment (SandroSal) [inline routes/api/landing.js]: is this function needed?

File: routes/api/landing.js (hunk 2 above)
Comment (SandroSal) [reply routes/api/landing.js]: sometimes we show 3, sometimes 4, sometimes 5 pictures under "Featured Cities"

File: front_end/src/actions/landingActions.js [hunk 4]
```diff
@@ -1,9 +1,19 @@
-import { GET_LANDING_DATA, SET_LANDING_STATUS } from "./types";
+import axios from "axios";
 
-export const readyLanding = () => {
-  return {
-    type: GET_LANDING_DATA
-  };
+import { GET_LANDING_DATA, SET_LANDING_STATUS, GET_ERRORS } from "./types";
```
[the remark this answers is not in the document]
Comment (DaleCS) [reply front_end/src/actions/landingActions.js]: Removed

File: front_end/src/components/landing_page/Landing.js [hunk 5]
```diff
@@ -10,17 +10,9 @@ import "./Landing.css";
 
 // Image imports (For Prototype Only)
 import SF from "./SF.jpg";
```
[the remark this answers is not in the document]
Comment (DaleCS) [reply front_end/src/components/landing_page/Landing.js]: Removed

File: routes/api/landing.js (hunk 2 above)
Comment (DaleCS) [reply routes/api/landing.js]: Will be done in different story.

File: routes/api/landing.js (hunk 3 above)
Comment (DaleCS) [reply routes/api/landing.js]: Will be done in different story. (Function will be refactored)

**Pull request merged**